ibis/expr/types/relations.py

```diff
@@ -2481,6 +2484,314 @@ def __exit__(self, exc_type, exc_val, exc_tb):
+        │ Jewish                  │    19 │      19 │      25 │      25 │      30 │ … │
+        │ …                       │     … │       … │       … │       … │       … │ … │
+        └─────────────────────────┴───────┴─────────┴─────────┴─────────┴─────────┴───┘
+        >>> relig_income.pivot_longer(~s.c("religion"), names_to="income", values_to="count")
+        ┏━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━┳━━━━━━━┓
+        ┃ religion ┃ income             ┃ count ┃
+        ┡━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━╇━━━━━━━┩
+        │ string   │ string             │ int64 │
+        ├──────────┼────────────────────┼───────┤
+        │ Agnostic │ <$10k              │    27 │
+        │ Agnostic │ $10-20k            │    34 │
+        │ Agnostic │ $20-30k            │    60 │
+        │ Agnostic │ $30-40k            │    81 │
+        │ Agnostic │ $40-50k            │    76 │
+        │ Agnostic │ $50-75k            │   137 │
+        │ Agnostic │ $75-100k           │   122 │
+        │ Agnostic │ $100-150k          │   109 │
+        │ Agnostic │ >150k              │    84 │
+        │ Agnostic │ Don't know/refused │    96 │
+        │ …        │ …                  │     … │
+        └──────────┴────────────────────┴───────┘
+        >>> world_bank_pop = ibis.examples.world_bank_pop_raw.fetch(header=1)
+        >>> world_bank_pop.head()
+        ┏━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━┓
+        ┃ country ┃ indicator   ┃ 2000         ┃ 2001         ┃ 2002         ┃ … ┃
+        ┡━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━┩
+        │ string  │ string      │ float64      │ float64      │ float64      │ … │
+        ├─────────┼─────────────┼──────────────┼──────────────┼──────────────┼───┤
+        │ ABW     │ SP.URB.TOTL │ 4.244400e+04 │ 4.304800e+04 │ 4.367000e+04 │ … │
+        │ ABW     │ SP.URB.GROW │ 1.182632e+00 │ 1.413021e+00 │ 1.434560e+00 │ … │
+        │ ABW     │ SP.POP.TOTL │ 9.085300e+04 │ 9.289800e+04 │ 9.499200e+04 │ … │
+        │ ABW     │ SP.POP.GROW │ 2.055027e+00 │ 2.225930e+00 │ 2.229056e+00 │ … │
+        │ AFG     │ SP.URB.TOTL │ 4.436299e+06 │ 4.648055e+06 │ 4.892951e+06 │ … │
+        └─────────┴─────────────┴──────────────┴──────────────┴──────────────┴───┘
+        >>> world_bank_pop.pivot_longer(s.matches(r"\\d{4}"), names_to="year").head()
+        ┏━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━┓
+        ┃ country ┃ indicator   ┃ year   ┃ value   ┃
+        ┡━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━┩
+        │ string  │ string      │ string │ float64 │
+        ├─────────┼─────────────┼────────┼─────────┤
+        │ ABW     │ SP.URB.TOTL │ 2000   │ 42444.0 │
+        │ ABW     │ SP.URB.TOTL │ 2001   │ 43048.0 │
+        │ ABW     │ SP.URB.TOTL │ 2002   │ 43670.0 │
+        │ ABW     │ SP.URB.TOTL │ 2003   │ 44246.0 │
+        │ ABW     │ SP.URB.TOTL │ 2004   │ 44669.0 │
+        └─────────┴─────────────┴────────┴─────────┘
```

Comment:
Would you mind adding a sentence or two describing what's done in the two examples above. This is both to provide more context into what `pivot_longer` is doing, and to also add a break between the two examples above, which kinda blend together without a sentence to break them apart.

Reply:
Done, filled out the rest of the examples with more prose.